supabase/migrations/20241210045518_update_pricing_logic.sql

```diff
@@ -0,0 +1,29 @@
+alter table "public"."proofs" drop column "proving_cost";
+
+drop view if exists "public"."recent_summary";
+
+create or replace view "public"."recent_summary" with (security_invoker=on) as
+  select 
+    count(distinct "b"."block_number") as "total_proven_blocks",
+    coalesce(avg((("cc"."instance_count" * "a"."hourly_price") * "p"."proving_time") / (1000.0 * 60 * 60)), 0::numeric) as "avg_cost_per_proof",
+    coalesce(avg("p"."proving_time"), 0::numeric) as "avg_proving_time"
+  from "public"."blocks" "b"
+  join "public"."proofs" "p" on "b"."block_number" = "p"."block_number" and "p"."proof_status" = 'proved'
+  join "public"."cluster_configurations" "cc" on "p"."cluster_id" = "cc"."cluster_id"
+  join "public"."aws_instance_pricing" "a" on "cc"."instance_type_id" = "a"."id"
+  where "b"."timestamp" >= (now() - '30 days'::interval);
+
+drop view if exists "public"."teams_summary";
+
+create or replace view "public"."teams_summary" with (security_invoker=on) as
+  select "t"."team_id",
+         "t"."team_name",
+         "t"."logo_url",
+         coalesce(sum(cc."instance_count" * a."hourly_price" * (p."proving_time" / (1000.0 * 60 * 60))) / nullif(count(p."proof_id"), 0), 0) as "avg_cost_per_proof",
```

Comment:
yeah, I think we need to re-think this. We need to calc the `avg(cost_per_proof)`. So, we first need to calculate the individual cost for each proof and then, the avg of those per cluster per team..

I don't think we will be able to do this in just one select...will need to undercover old sql books xD

Reply:
@pettinarip

Okay, so we're trying to calculate the **Average Cost Per Proof**, which = **total costs** / **total number of proofs**

Below we have `and "p"."proof_status" = 'proved'` to filter out proofs not yet finished.

We're summing up the products of three things:
- cc."instance_count"
- a."hourly_price"
- And p."proving_time" (with ms/hours adjustment)

This gives us the **total costs**. We don't need to `sum(instance_count*hourly_price)` and _then_ multiply by `proving_time`, we can also just multiply each `instance_count*hourly_price` by `proving_time` and sum those... same mathematical result.

Then our **total number of proofs** is just `count(p."proof_id")`, but we use `nullif(count(p."proof_id"), 0)` to make sure we're not dividing by zero.

I think this works.

Reply:
Example:

```
1 * 5 = 5
2 * 4 = 8
3 * 3 = 9
---------
       22 * 10 = 220

```


```
1 * 5 * 10 = 50
2 * 4 * 10 = 80
3 * 3 * 10 = 90
---------------
            220
```

Reply:
yeah, I think this looks fine! I got confused on the how we were getting the cost per machine.